lib/core/src/Cardano/Wallet/DB/Model.hs

```diff
@@ -322,6 +346,23 @@ mRollbackTo wid requested db@(Database wallets txs) = case Map.lookup wid wallet
     safeHead [] = Nothing
     safeHead (h:_) = Just h
 
+mPrune :: Ord wid => wid -> Quantity "block" Word32 -> ModelOp wid s xprv ()
+mPrune wid epochStability = alterModel wid $ ((),) . prune
+  where
+    prune wal = wal
+        { checkpoints = Map.filter ((<= point) . cpHeight) $
+            checkpoints wal
+        , submittedTxs = Map.filterWithKey (const . ((<= point) . txHeight)) $
+            submittedTxs wal
+        }
+      where
+        cp = mostRecentCheckpoint wal
+        tipHeight = maybe 0 (getQuantity . cpHeight) cp
+        point = Quantity (tipHeight - getQuantity epochStability)
+        cpHeight = view (#currentTip . #blockHeight)
+        txHeight txid = maybe (Quantity tipHeight) (view #blockHeight) $
+            Map.lookup txid $ txHistory wal
```

Comment:
That isn't how the pruning works currently with SQLite. This isn't caught by the SM testing however because we never generate prune commands, but if we were I suspect that it'd fail :sweat_smile: ... 

We do prune the SQLite database using the `sparseCheckpoints` function, to reduce the number of past checkpoints we have archived. Pruning should likely not have any impact on submitted transactions however; and although interesting for testing & al, I think defining a model version for `prune` is out of the scope of this PR :thinking:  ?

Reply:
Ok yes, I have reverted that and intend to address it in another PR.